### PR TITLE
changed settingsrepo backend to use database instead of Settings

### PR DESCRIPTION
**Pull request by helloseyedjafari**
Previously we were using [Settings](https://github.com/russhwolf/multiplatform-settings) as our saving solution to store and persist the key/values.
**settings** had a limitation on observing the values. 
SQLDelight has builtin observers so I decided to use our database instead of the Settings library.

the only downside is that we have to manually cast everything back from strings to the types we need. since we are only going to store primitives this should not be a big problem.

make sure to clear your data when installing this branch

### CODE REVIEW

File: core/src/commonMain/kotlin/com/islamversity/core/Logger.kt [hunk 1]
```diff
@@ -0,0 +1,63 @@
+package com.islamversity.core
+
+import co.touchlab.kermit.CommonLogger
+import co.touchlab.kermit.Kermit
+import co.touchlab.kermit.Logger
+import com.islamversity.core.Logger.init
+import co.touchlab.kermit.Severity as KSeverity
+
+enum class Severity {
```
Comment (rezaiyan) [inline core/src/commonMain/kotlin/com/islamversity/core/Logger.kt:9]: 😰 enum

**Review by rezaiyan — APPROVED** [review]
Well done

File: core/src/commonMain/kotlin/com/islamversity/core/Logger.kt (hunk 1 above)
Comment (helloseyedjafari) [reply core/src/commonMain/kotlin/com/islamversity/core/Logger.kt:9]: why not?

Comment (helloseyedjafari) [reply core/src/commonMain/kotlin/com/islamversity/core/Logger.kt:9]: you mean I should create functions instead?

Comment (rezaiyan) [reply core/src/commonMain/kotlin/com/islamversity/core/Logger.kt:9]: You know, a single enum is 13x bigger than the Integer constant

**Pull request merged**